source/Timer.h

```diff
@@ -1,285 +1,72 @@
+    Timer &operator=(Timer &&) = delete;
 
-        /**
-         * @brief Creates and returns a new QUIT message.
-         */
-        static TimerMessage *createQuitMessage() { return new TimerMessage(TIMER_MSG_QUIT); }
+    void cancel() override;
+    bool isActive() const override;
 
-        /**
-         * @brief Virtual destructor.
-         */
-        virtual ~TimerMessage(){};
-
-        /**
-         * @brief Returns the message type.
-         *
-         * @retval the message type.
-         */
-        MessageType getType() { return m_type; };
-
-        /**
-         * @brief Returns the message timeout value.
-         *
-         * @retval the message timeout value.
-         */
-        long getTimeout() { return m_timeout; };
-
-    protected:
-        /**
-         * @brief The message type.
-         */
-        MessageType m_type;
-
-        /**
-         * @brief The timeout.
-         */
-        long m_timeout;
-
-        /**
-         * @brief Constructor.
-         *
-         * @param[in] type    : The message type.
-         * @param[in] timeout : The timeout (default -1)
-         */
-        TimerMessage(MessageType type = TIMER_MSG_UNKNOWN, long timeout = -1) : m_type(type), m_timeout(timeout){};
-    };
-
-    /**
-     * @brief The callback function
-     */
-    TimerCallback m_callbackFunction;
-
-    /**
-     * @brief The user data
-     */
-    void *m_userData;
-
-    /**
-     * @brief The message queue protection mutex.
-     */
-    pthread_mutex_t m_mutex;
-
-    /**
-     * @brief The message queue protection mutex attributes.
-     */
-    pthread_mutexattr_t m_mutexAttr;
-
-    /**
-     * @brief The message queue condition variable.
-     */
-    pthread_cond_t m_cond;
-
-    /**
-     * @brief The message queue condition variable attibutes.
-     */
-    pthread_condattr_t m_condAttr;
-
-    /**
-     * @brief The message queue.
-     */
-    std::deque<TimerMessage *> m_msgQueue;
-
-    /**
-     * @brief The timer thread.
-     */
-    pthread_t m_timerThread;
-
-    /**
-     * @brief The timer thread static function.
-     *
-     * @param[in] self : A void 'this' pointer.
-     *
-     * @retval not NULL if exit without error.
-     */
-    static void *timerThreadFunction(void *self);
-
-    /**
-     * @brief The timer thread function.
-     *
-     * @retval true if exit without error.
-     */
-    bool timerThread();
-
-    /**
-     * @brief Posts a timer message to the timer thread.
-     *
-     * This method posts a timer message to the timer thread. Set urgent to
-     * true to post a message to the front of the queue. I.e. ensure it is
-     * the next message handled.
-     *
-     * @param[in] message : The timer message.
-     * @param[in] urgent  : The urgency flag
-     */
-    void postTimerMessage(TimerMessage *message, bool urgent = false);
+private:
+    void doCancel();
 
-    /**
-     * @brief Timer name.
-     */
-    std::string m_name;
+private:
```

Comment:
I think you may have added this access modifier by mistake, don't think it's needed